rust/lib.rs

```diff
@@ -88,20 +100,24 @@ impl Schema {
     pub fn validate_json(&self, input: &[u8]) -> Result<(), SchemaError> {
         let message = serde_json::from_slice(input).map_err(|_| SchemaError::InvalidMessage)?;
 
-        if let Some(compiled) = &self.compiled_json_schema {
-            compiled
-                .validate(&message)
-                .map_err(|_| SchemaError::InvalidMessage)
-        } else {
-            Err(SchemaError::InvalidSchema)
-        }
+        let compiled = self
+            .compiled_json_schema
+            .as_ref()
+            .ok_or(SchemaError::InvalidSchema)?;
+
+        compiled
+            .validate(&message)
+            .map_err(|_| SchemaError::InvalidMessage)?;
+
+        // FIXME(swatinem): we might as well just return the `Value` here
```

Comment:
i guess we can do that but i feel like it's not useful, also i hoped to shop around for a faster json schema validator at some point and get rid of creating a Value

Reply:
if we are okay with breaking the API more frequently, we can add this now and remove that later on if we find a more efficient solution. according to my benchmarks, JSON parsing is one of the most expensive parts of the consumers, and we would be doing it twice otherwise. Properly hooking this up might be a challenge though.

Reply:
either option sounds fine to me, the API does not have to be very stable, and I think it's unlikely to break any code since I believe most users won't care for this return value eitehr